Could not pass event PlayerRightClickEvent to Slimefun

### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

```
com.destroystokyo.paper.exception.ServerEventException: Could not pass event PlayerRightClickEvent to Slimefun vDEV - 999 (git e02eedd9)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:647)
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.listeners.SlimefunItemInteractListener.onRightClick(SlimefunItemInteractListener.java:62)
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor375.execute(Unknown Source)
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:543)
	at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:542)
	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1815)
	at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:33)
	at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:9)
	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$1(PlayerConnectionUtils.java:56)
	at net.minecraft.server.TickTask.run(TickTask.java:18)
	at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:149)
	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(IAsyncTaskHandlerReentrant.java:23)
	at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:1426)
	at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:192)
	at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:122)
	at net.minecraft.server.MinecraftServer.bf(MinecraftServer.java:1404)
	at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1397)
	at net.minecraft.util.thread.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:132)
	at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1375)
	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1286)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because "customName" is null
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.items.altar.AncientPedestal.getOriginalItemStack(AncientPedestal.java:112)
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.listeners.AncientAltarListener.usePedestal(AncientAltarListener.java:159)
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.listeners.AncientAltarListener.onInteract(AncientAltarListener.java:111)
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor440.execute(Unknown Source)
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
	at Slimefun4.jar//io.github.thebusybiscuit.slimefun4.implementation.listeners.SlimefunItemInteractListener.onRightClick(SlimefunItemInteractListener.java:62)
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor375.execute(Unknown Source)
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628)
	at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:543)
	at app//net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:542)
	at app//net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1815)
	at app//net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:33)
	at app//net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.a(ServerboundUseItemOnPacket.java:9)
	at app//net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:56)
	at app//net.minecraft.server.TickTask.run(TickTask.java:18)
	at app//net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:149)
	at app//net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23)
	at app//net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1426)
	at app//net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:192)
	at app//net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122)
	at app//net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1404)
	at app//net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1397)
	at app//net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:132)
	at app//net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1375)
	at app//net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1286)
	... 2 more
```

### 📑 Reproduction Steps

Not sure.

### 💡 Expected Behavior

N/A

### 📷 Screenshots / Videos

_No response_

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.17.x

### ⭐ Slimefun version

```
> sf versions
[03:04:41 INFO]: This Server uses the following setup of Slimefun:
Paper git-Paper-408 (MC: 1.17.1)
Slimefun DEV - 999 (git e02eedd9)
Metrics-Module #28
Java 17

Installed Addons: (3)
  EcoPower vDEV - 5 (git 5c1ecd68)
  SlimefunLuckyBlocks vDEV - 36 (git 60f0c8dc)
  ExoticGarden vDEV - 77 (git a2c4b6d3)
```

### 🧭 Other plugins

_No response_

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -131,8 +131,8 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
     }
     
     private boolean testArmorStand(@Nullable Entity n) {
-        if (n instanceof ArmorStand armorStand && armorStand.isValid()) {
-            String customName = armorStand.getCustomName();
+        if (n instanceof ArmorStand && n.isValid()) {
+            String customName = n.getCustomName();
             return customName != null && customName.startsWith(ITEM_PREFIX);
         } else {
             return false;

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -6,8 +6,6 @@ import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import javax.annotation.ParametersAreNonnullByDefault;
 
-import io.github.thebusybiscuit.slimefun4.core.attributes.NotHopperable;
-import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import org.bukkit.GameMode;
 import org.bukkit.Location;
 import org.bukkit.Sound;
@@ -27,6 +25,7 @@ import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
 import io.github.thebusybiscuit.slimefun4.api.items.ItemSpawnReason;
 import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
 import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.NotHopperable;
 import io.github.thebusybiscuit.slimefun4.core.handlers.BlockBreakHandler;
 import io.github.thebusybiscuit.slimefun4.core.handlers.BlockDispenseHandler;
 import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
@@ -34,6 +33,7 @@ import io.github.thebusybiscuit.slimefun4.implementation.handlers.SimpleBlockBre
 import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
 import io.github.thebusybiscuit.slimefun4.implementation.listeners.AncientAltarListener;
 import io.github.thebusybiscuit.slimefun4.implementation.tasks.AncientAltarTask;
+import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
 
 /**

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -181,6 +181,8 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
             entity.setVelocity(new Vector(0, 0.1, 0));
             entity.setCustomNameVisible(true);
             entity.setCustomName(nametag);
+            armorStand.setCustomName(nametag);
+            armorStand.setCustomNameVisible(false);
             armorStand.addPassenger(entity);
             SlimefunUtils.markAsNoPickup(entity, "altar_item");
             p.playSound(b.getLocation(), Sound.ENTITY_ITEM_PICKUP, 0.3F, 0.3F);

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -6,10 +6,13 @@ import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import io.github.thebusybiscuit.slimefun4.core.attributes.NotHopperable;
+import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import org.bukkit.GameMode;
 import org.bukkit.Location;
 import org.bukkit.Sound;
 import org.bukkit.block.Block;
+import org.bukkit.entity.ArmorStand;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.Item;
 import org.bukkit.entity.Player;
@@ -40,13 +43,14 @@ import io.github.thebusybiscuit.slimefun4.utils.SlimefunUtils;
  * 
  * @author Redemption198
  * @author TheBusyBiscuit
+ * @author JustAHuman
  * 
  * @see AncientAltar
  * @see AncientAltarListener
  * @see AncientAltarTask
  *
  */
-public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> {
+public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> implements NotHopperable {
 
     public static final String ITEM_PREFIX = ChatColors.color("&dALTAR &3Probe - &e");
 
@@ -63,7 +67,8 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> {
             @Override
             public void onBlockBreak(@Nonnull Block b) {
                 Optional<Item> entity = getPlacedItem(b);
-
+                Optional<ArmorStand> armorStand = getArmorStand(b);
+                
                 if (entity.isPresent()) {
                     Item stack = entity.get();
 
@@ -71,6 +76,14 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> {
                         stack.removeMetadata("no_pickup", Slimefun.instance());
                         b.getWorld().dropItem(b.getLocation(), getOriginalItemStack(stack));
                         stack.remove();
+                    }
+                }
+                
+                if (armorStand.isPresent()) {
+                    ArmorStand stand = armorStand.get();
+                    
+                    if (stand.isValid()) {
+                        stand.remove();
                     }
                 }
             }
@@ -93,12 +106,38 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> {
 
         return Optional.empty();
     }
+    
+    public @Nonnull Optional<ArmorStand> getArmorStand(@Nonnull Block pedestal) {
+        Optional<Item> entity = getPlacedItem(pedestal);
+        
+        if (entity.isPresent() && entity.get().getVehicle() instanceof ArmorStand armorStand) {
+            return Optional.of(armorStand);
+        }
+    
+        Location l = pedestal.getLocation().add(0.5, 1.2, 0.5);
+        for (Entity n : l.getWorld().getNearbyEntities(l, 0.5, 0.5, 0.5, this::testArmorStand)) {
+            if (n instanceof ArmorStand armorStand) {
+                return Optional.of(armorStand);
+            }
+        }
+        
+        return Optional.empty();
+    }
 
     private boolean testItem(@Nullable Entity n) {
         if (n instanceof Item item && n.isValid()) {
             ItemMeta meta = item.getItemStack().getItemMeta();
 
             return meta.hasDisplayName() && meta.getDisplayName().startsWith(ITEM_PREFIX);
+        } else {
+            return false;
+        }
+    }
+    
+    private boolean testArmorStand(@Nullable Entity n) {
+        if (n instanceof ArmorStand armorStand && armorStand.isValid()) {
+            String customName = armorStand.getCustomName();
+            return customName != null && customName.startsWith(ITEM_PREFIX);
         } else {
             return false;
         }
@@ -136,9 +175,13 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> {
         Item entity = SlimefunUtils.spawnItem(b.getLocation().add(0.5, 1.2, 0.5), displayItem, ItemSpawnReason.ANCIENT_PEDESTAL_PLACE_ITEM);
 
         if (entity != null) {
+            ArmorStand armorStand = getArmorStand(b).orElse(ArmorStandUtils.spawnArmorStand(b.getLocation().add(0.5, 1.2, 0.5)));
+            entity.setInvulnerable(true);
+            entity.setUnlimitedLifetime(true);
             entity.setVelocity(new Vector(0, 0.1, 0));
             entity.setCustomNameVisible(true);
             entity.setCustomName(nametag);
+            armorStand.addPassenger(entity);
             SlimefunUtils.markAsNoPickup(entity, "altar_item");
             p.playSound(b.getLocation(), Sound.ENTITY_ITEM_PICKUP, 0.3F, 0.3F);
         }

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -67,7 +67,7 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
             @Override
             public void onBlockBreak(@Nonnull Block b) {
                 Optional<Item> entity = getPlacedItem(b);
-                Optional<ArmorStand> armorStand = getArmorStand(b);
+                ArmorStand armorStand = getArmorStand(b, false);
                 
                 if (entity.isPresent()) {
                     Item stack = entity.get();
@@ -79,12 +79,8 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
                     }
                 }
                 
-                if (armorStand.isPresent()) {
-                    ArmorStand stand = armorStand.get();
-                    
-                    if (stand.isValid()) {
-                        stand.remove();
-                    }
+                if (armorStand != null && armorStand.isValid()) {
+                    armorStand.remove();
                 }
             }
         };
@@ -107,21 +103,21 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
         return Optional.empty();
     }
     
-    public @Nonnull Optional<ArmorStand> getArmorStand(@Nonnull Block pedestal) {
+    public @Nullable ArmorStand getArmorStand(@Nonnull Block pedestal, boolean createIfNoneExists) {
         Optional<Item> entity = getPlacedItem(pedestal);
         
         if (entity.isPresent() && entity.get().getVehicle() instanceof ArmorStand armorStand) {
-            return Optional.of(armorStand);
+            return armorStand;
         }
     
         Location l = pedestal.getLocation().add(0.5, 1.2, 0.5);
         for (Entity n : l.getWorld().getNearbyEntities(l, 0.5, 0.5, 0.5, this::testArmorStand)) {
             if (n instanceof ArmorStand armorStand) {
-                return Optional.of(armorStand);
+                return armorStand;
             }
         }
         
-        return Optional.empty();
+        return createIfNoneExists ? ArmorStandUtils.spawnArmorStand(l) : null;
     }
 
     private boolean testItem(@Nullable Entity n) {
@@ -162,7 +158,8 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
 
     public void placeItem(@Nonnull Player p, @Nonnull Block b) {
         ItemStack hand = p.getInventory().getItemInMainHand();
-        ItemStack displayItem = new CustomItemStack(hand, ITEM_PREFIX + System.nanoTime());
+        String displayName = ITEM_PREFIX + System.nanoTime();
+        ItemStack displayItem = new CustomItemStack(hand, displayName);
         displayItem.setAmount(1);
 
         // Get the display name of the original Item in the Player's hand
@@ -175,14 +172,13 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
         Item entity = SlimefunUtils.spawnItem(b.getLocation().add(0.5, 1.2, 0.5), displayItem, ItemSpawnReason.ANCIENT_PEDESTAL_PLACE_ITEM);
 
         if (entity != null) {
-            ArmorStand armorStand = getArmorStand(b).orElse(ArmorStandUtils.spawnArmorStand(b.getLocation().add(0.5, 1.2, 0.5)));
+            ArmorStand armorStand = getArmorStand(b, true);
             entity.setInvulnerable(true);
             entity.setUnlimitedLifetime(true);
             entity.setVelocity(new Vector(0, 0.1, 0));
             entity.setCustomNameVisible(true);
             entity.setCustomName(nametag);
-            armorStand.setCustomName(nametag);
-            armorStand.setCustomNameVisible(false);
+            armorStand.setCustomName(displayName);
             armorStand.addPassenger(entity);
             SlimefunUtils.markAsNoPickup(entity, "altar_item");
             p.playSound(b.getLocation(), Sound.ENTITY_ITEM_PICKUP, 0.3F, 0.3F);

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -141,14 +141,12 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
 
     public @Nonnull ItemStack getOriginalItemStack(@Nonnull Item item) {
         ItemStack stack = item.getItemStack().clone();
+        ItemMeta im = stack.getItemMeta();
         String customName = item.getCustomName();
+        im.setDisplayName(null);
+        stack.setItemMeta(im);
 
-        if (customName != null && customName.equals(ItemUtils.getItemName(new ItemStack(stack.getType())))) {
-            ItemMeta im = stack.getItemMeta();
-            im.setDisplayName(null);
-            stack.setItemMeta(im);
-        } else {
-            ItemMeta im = stack.getItemMeta();
+        if (customName == null || !customName.equals(ItemUtils.getItemName(stack))) {
             im.setDisplayName(customName);
             stack.setItemMeta(im);
         }

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/altar/AncientPedestal.java
@@ -143,7 +143,7 @@ public class AncientPedestal extends SimpleSlimefunItem<BlockDispenseHandler> im
         ItemStack stack = item.getItemStack().clone();
         String customName = item.getCustomName();
 
-        if (customName.equals(ItemUtils.getItemName(new ItemStack(stack.getType())))) {
+        if (customName != null && customName.equals(ItemUtils.getItemName(new ItemStack(stack.getType())))) {
             ItemMeta im = stack.getItemMeta();
             im.setDisplayName(null);
             stack.setItemMeta(im);

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
@@ -9,7 +9,6 @@ import org.bukkit.Material;
 import org.bukkit.block.Block;
 import org.bukkit.entity.ArmorStand;
 import org.bukkit.entity.Entity;
-import org.bukkit.entity.EntityType;
 import org.bukkit.entity.Player;
 import org.bukkit.event.block.BlockPlaceEvent;
 import org.bukkit.inventory.ItemStack;
@@ -155,6 +154,7 @@ public class HologramProjector extends SlimefunItem implements HologramOwner {
         }
 
         ArmorStand hologram = ArmorStandUtils.spawnArmorStand(l);
+        hologram.setCustomNameVisible(true);
         hologram.setCustomName(nametag);
         return hologram;
     }

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
@@ -3,6 +3,7 @@ package io.github.thebusybiscuit.slimefun4.implementation.items.blocks;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.block.Block;
@@ -153,21 +154,9 @@ public class HologramProjector extends SlimefunItem implements HologramOwner {
             return null;
         }
 
-        ArmorStand hologram = spawnArmorStand(l);
+        ArmorStand hologram = ArmorStandUtils.spawnArmorStand(l);
         hologram.setCustomName(nametag);
         return hologram;
-    }
-
-    private static @Nonnull ArmorStand spawnArmorStand(@Nonnull Location l) {
-        ArmorStand armorStand = (ArmorStand) l.getWorld().spawnEntity(l, EntityType.ARMOR_STAND);
-        armorStand.setVisible(false);
-        armorStand.setSilent(true);
-        armorStand.setMarker(true);
-        armorStand.setGravity(false);
-        armorStand.setBasePlate(false);
-        armorStand.setCustomNameVisible(true);
-        armorStand.setRemoveWhenFarAway(false);
-        return armorStand;
     }
 
     private static void killArmorStand(@Nonnull Block b) {

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
@@ -3,7 +3,6 @@ package io.github.thebusybiscuit.slimefun4.implementation.items.blocks;
 import javax.annotation.Nonnull;
 import javax.annotation.ParametersAreNonnullByDefault;
 
-import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.block.Block;
@@ -26,6 +25,7 @@ import io.github.thebusybiscuit.slimefun4.core.handlers.BlockUseHandler;
 import io.github.thebusybiscuit.slimefun4.core.services.holograms.HologramsService;
 import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
 import io.github.thebusybiscuit.slimefun4.implementation.handlers.SimpleBlockBreakHandler;
+import io.github.thebusybiscuit.slimefun4.utils.ArmorStandUtils;
 import io.github.thebusybiscuit.slimefun4.utils.ChatUtils;
 import io.github.thebusybiscuit.slimefun4.utils.NumberUtils;
 

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java
@@ -152,11 +152,8 @@ public class HologramProjector extends SlimefunItem implements HologramOwner {
         if (!createIfNoneExists) {
             return null;
         }
-
-        ArmorStand hologram = ArmorStandUtils.spawnArmorStand(l);
-        hologram.setCustomNameVisible(true);
-        hologram.setCustomName(nametag);
-        return hologram;
+        
+        return ArmorStandUtils.spawnArmorStand(l, nametag);
     }
 
     private static void killArmorStand(@Nonnull Block b) {

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -1,12 +1,13 @@
 package io.github.thebusybiscuit.slimefun4.utils;
 
-import io.github.thebusybiscuit.slimefun4.core.services.holograms.HologramsService;
-import io.github.thebusybiscuit.slimefun4.implementation.items.altar.AncientPedestal;
-import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.HologramProjector;
+import javax.annotation.Nonnull;
+
 import org.bukkit.Location;
 import org.bukkit.entity.ArmorStand;
 
-import javax.annotation.Nonnull;
+import io.github.thebusybiscuit.slimefun4.core.services.holograms.HologramsService;
+import io.github.thebusybiscuit.slimefun4.implementation.items.altar.AncientPedestal;
+import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.HologramProjector;
 
 /**
  * This class holds utilities for {@link ArmorStand}, useful for classes

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -1,0 +1,39 @@
+package io.github.thebusybiscuit.slimefun4.utils;
+
+import io.github.thebusybiscuit.slimefun4.core.services.holograms.HologramsService;
+import io.github.thebusybiscuit.slimefun4.implementation.items.altar.AncientPedestal;
+import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.HologramProjector;
+import org.bukkit.Location;
+import org.bukkit.entity.ArmorStand;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class holds utilities for {@link ArmorStand}, useful for classes
+ * dealing with {@link ArmorStand}s that are not from {@link HologramsService}
+ *
+ * @see HologramProjector
+ * @see AncientPedestal
+ *
+ * @author JustAHuman
+ */
+public class ArmorStandUtils {
+    /**
+     * Spawns an {@link ArmorStand} at the given {@link Location}
+     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, Visible CustomName, NoRemoveWhenFarAway)
+     *
+     * @param location The {@link Location} to spawn the {@link ArmorStand}
+     * @return The spawned {@link ArmorStand}
+     */
+    public static ArmorStand spawnArmorStand(@Nonnull Location location) {
+        ArmorStand armorStand = location.getWorld().spawn(location, ArmorStand.class);
+        armorStand.setVisible(false);
+        armorStand.setSilent(true);
+        armorStand.setMarker(true);
+        armorStand.setGravity(false);
+        armorStand.setBasePlate(false);
+        armorStand.setCustomNameVisible(true);
+        armorStand.setRemoveWhenFarAway(false);
+        return armorStand;
+    }
+}

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -22,7 +22,8 @@ public class ArmorStandUtils {
 
     /**
      * Spawns an {@link ArmorStand} at the given {@link Location} with the given custom name
-     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway, VisibleCustomName)
+     * <br>
+     * Set Properties: Invisible, Silent, Marker, No-Gravity, No Base Plate, Don't Remove When Far Away
      *
      * @param location The {@link Location} to spawn the {@link ArmorStand}
      * @param customName The {@link String} custom name the {@link ArmorStand} should display
@@ -38,7 +39,8 @@ public class ArmorStandUtils {
     
     /**
      * Spawns an {@link ArmorStand} at the given {@link Location}
-     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway)
+     * <br>
+     * Set Properties: Invisible, Silent, Marker, No-Gravity, No Base Plate, Don't Remove When Far Away
      *
      * @param location The {@link Location} to spawn the {@link ArmorStand}
      *

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -19,6 +19,7 @@ import io.github.thebusybiscuit.slimefun4.implementation.items.blocks.HologramPr
  * @author JustAHuman
  */
 public class ArmorStandUtils {
+
     /**
      * Spawns an {@link ArmorStand} at the given {@link Location} with the given custom name
      * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway, VisibleCustomName)

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -28,8 +28,7 @@ public class ArmorStandUtils {
      *
      * @return The spawned {@link ArmorStand}
      */
-    @Nonnull
-    public static ArmorStand spawnArmorStand(@Nonnull Location location, @Nonnull String customName) {
+    public static @Nonnull ArmorStand spawnArmorStand(@Nonnull Location location, @Nonnull String customName) {
         ArmorStand armorStand = spawnArmorStand(location);
         armorStand.setCustomName(customName);
         armorStand.setCustomNameVisible(true);
@@ -44,8 +43,7 @@ public class ArmorStandUtils {
      *
      * @return The spawned {@link ArmorStand}
      */
-    @Nonnull
-    public static ArmorStand spawnArmorStand(@Nonnull Location location) {
+    public static @Nonnull ArmorStand spawnArmorStand(@Nonnull Location location) {
         return location.getWorld().spawn(location, ArmorStand.class, armorStand -> {
             armorStand.setVisible(false);
             armorStand.setSilent(true);

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -19,12 +19,31 @@ import javax.annotation.Nonnull;
  */
 public class ArmorStandUtils {
     /**
+     * Spawns an {@link ArmorStand} at the given {@link Location} with the given custom name
+     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway, VisibleCustomName)
+     *
+     * @param location The {@link Location} to spawn the {@link ArmorStand}
+     * @param customName The {@link String} custom name the {@link ArmorStand} should display
+     *
+     * @return The spawned {@link ArmorStand}
+     */
+    @Nonnull
+    public static ArmorStand spawnArmorStand(@Nonnull Location location, @Nonnull String customName) {
+        ArmorStand armorStand = spawnArmorStand(location);
+        armorStand.setCustomName(customName);
+        armorStand.setCustomNameVisible(true);
+        return armorStand;
+    }
+    
+    /**
      * Spawns an {@link ArmorStand} at the given {@link Location}
      * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway)
      *
      * @param location The {@link Location} to spawn the {@link ArmorStand}
+     *
      * @return The spawned {@link ArmorStand}
      */
+    @Nonnull
     public static ArmorStand spawnArmorStand(@Nonnull Location location) {
         ArmorStand armorStand = location.getWorld().spawn(location, ArmorStand.class);
         armorStand.setVisible(false);

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -20,7 +20,7 @@ import javax.annotation.Nonnull;
 public class ArmorStandUtils {
     /**
      * Spawns an {@link ArmorStand} at the given {@link Location}
-     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, Visible CustomName, NoRemoveWhenFarAway)
+     * (Invisible, Silent, Marker, NoGravity, NoBasePlate, NoRemoveWhenFarAway)
      *
      * @param location The {@link Location} to spawn the {@link ArmorStand}
      * @return The spawned {@link ArmorStand}
@@ -32,7 +32,6 @@ public class ArmorStandUtils {
         armorStand.setMarker(true);
         armorStand.setGravity(false);
         armorStand.setBasePlate(false);
-        armorStand.setCustomNameVisible(true);
         armorStand.setRemoveWhenFarAway(false);
         return armorStand;
     }

--- a/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
+++ b/src/main/java/io/github/thebusybiscuit/slimefun4/utils/ArmorStandUtils.java
@@ -46,13 +46,13 @@ public class ArmorStandUtils {
      */
     @Nonnull
     public static ArmorStand spawnArmorStand(@Nonnull Location location) {
-        ArmorStand armorStand = location.getWorld().spawn(location, ArmorStand.class);
-        armorStand.setVisible(false);
-        armorStand.setSilent(true);
-        armorStand.setMarker(true);
-        armorStand.setGravity(false);
-        armorStand.setBasePlate(false);
-        armorStand.setRemoveWhenFarAway(false);
-        return armorStand;
+        return location.getWorld().spawn(location, ArmorStand.class, armorStand -> {
+            armorStand.setVisible(false);
+            armorStand.setSilent(true);
+            armorStand.setMarker(true);
+            armorStand.setGravity(false);
+            armorStand.setBasePlate(false);
+            armorStand.setRemoveWhenFarAway(false);
+        });
     }
 }

